src/exchange.ts

```diff
@@ -139,6 +139,8 @@ export const cacheExchange = (opts: CacheExchangeOpts): Exchange => ({
     const policy = getRequestPolicy(operation);
     const res = query(store, operation);
     const isComplete = policy === 'cache-only' || res.completeness === 'FULL';
+    res.dependencies = { ...res.dependencies, ...store.pendingDependencies };
+    store.pendingDependencies = new Set();
```

Comment:
What if we only used store dependencies? We'd have a reset method call here that creates a new set on the store and returns it, which we set on dependencies.

Then we call "mark Dependency" instead of accessing the set directly on the store.

We could also reset again after each operation to prevent these store methods from being used outside of operations

Reply:
I'm all for the helpers you are describing but only storing here would enforce us to make an "in-between" method in store to use `query`